openquake/hazardlib/site_amplification.py

```diff
@@ -131,15 +261,32 @@ def amplify_one(self, ampl_code, imt, poes):
         """
         if isinstance(poes, list):  # in the tests
             poes = numpy.array(poes).reshape(-1, 1)
+        # Manage the case of a site collection with empty ampcode
         if ampl_code == b'' and len(self.ampcodes) == 1:
-            # manage the case of a site collection with empty ampcode
             ampl_code = self.ampcodes[0]
-        ialphas = self.ialphas[ampl_code, imt]
-        isigmas = self.isigmas[ampl_code, imt]
+        # Get median amplification an std of the logarithm
+        if mag is None and dst is None:
+            ialphas = self.ialphas[ampl_code, imt]
+            isigmas = self.isigmas[ampl_code, imt]
+
         A, G = len(self.amplevels), poes.shape[1]
         ampl_poes = numpy.zeros((A, G))
+        #
+        # Amplify, for each site i.e. distance
         for g in range(G):
+            # Compute the probability of occurrence of GM within a number of
+            # intervals
             p_occ = -numpy.diff(poes[:, g])
+            #
+            # Setting the amplification function
+            if mag is not None and dst is not None:
+                1/0  # ERROR: it never goes here
```

Comment:
It never enters here because in openquake/calculators/classical.py the call is `pcurves = amplifier.amplify(ampcode[sid], pcurves)` without passing magnitude and rrup (which is correct for the convolution).